activerecord/lib/active_record/secure_token.rb

```diff
@@ -23,11 +23,14 @@ module ClassMethods
       # Note that it's still possible to generate a race condition in the database in the same way that
       # <tt>validates_uniqueness_of</tt> can. You're encouraged to add a unique index in the database to deal
       # with this even more unlikely scenario.
-      def has_secure_token(attribute = :token)
+      def has_secure_token(attribute = :token, options = {})
```

Comment:
Let's just pass the options straight through: `def has_secure_token(attribute = :token, **before_create_options)` and later `before_create(before_create_options) { ... }`.